### DISCUSSION TITLE
Enabling Elution-Centric Search with AlphaPept Features

### DISCUSSION THREAD

**Discussion by ec-ho-ra-mos** [MannLabs/alphapept#559] [Ideas]
Hello AlphaPept Team,

First, I want to commend you for your dedication to open-source science!

I'm interested in exploring the possibility of recovering the individual signals associated with each feature. This includes retrieving the intensity sum/area for each signal along with its corresponding retention time (RT) information. I believe this data would be valuable for non-spectrum-centric developers who want to leverage AlphaPept's feature detection algorithm to develop elution profile-centric search methods.
![image](https://github.com/MannLabs/alphapept/assets/61624696/d43131fe-8fcf-4dbd-a15c-092df6f2b5e1)

**Comment by straussmaximilian** [comment]
Hi yes,
there are two ways do to this:


1) Via AlphaPept
Here is a notebook for a test run:
The last cell show how you can get back to the isotope information from the feature finder data.

```
feature_mapping = ms_file.read(dataset_name='feature_cluster_mapping')
display(feature_mapping.head())
feature_table = ms_file.read(dataset_name='feature_table')
feature_table['isotope_mzs'] = feature_mapping.groupby('feature_id')['mz_average'].apply(list)
feature_table['isotope_ints'] = feature_mapping.groupby('feature_id')['int_sum'].apply(list)

df[['isotope_mzs','isotope_ints']] = feature_table.loc[df['feature_idx'].tolist()][['isotope_mzs','isotope_ints']].values

display(df[['sequence','ms1_int_max_apex','isotope_mzs','isotope_ints']])
```

https://github.com/MannLabs/alphapept/blob/master/sandbox/01_test_run.ipynb

2) Via AlphaRaw.
If you want to have just the raw conversion and feature finding you might want to check out AlphaRaw: We recently migrated the feature finder code here aswell.
https://github.com/MannLabs/alpharaw/blob/main/nbs/feature_finding.ipynb

**Reply by ec-ho-ra-mos** [reply]
Thank you very much, this is very helpful! 

In light of this, do you think it also possible to use the same "feature finding" algorithm for MS2 signals?  This would allow us to easily compare the elution profiles of individual MS2 peaks and see whether candidate fragment ion peaks align with each other, for scoring.
